src/common/regions.ts

```diff
@@ -116,27 +121,61 @@ const counties: County[] = chain(state_county_map_dataset)
 class RegionDB {
   constructor(private regions: Region[]) {}
 
-  private filterByType(regionType: RegionType) {
-    return this.all().filter(region => region.regionType === regionType);
-  }
-
   findByFipsCode(fipsCode: FipsCode): Region | null {
     const region = this.regions.find(region => region.fipsCode === fipsCode);
     return region || null;
   }
 
+  findStateByUrlParams(stateId: string): State | null {
+    const foundState = this.states().find(
+      state =>
+        equalLower(state.urlSegment, stateId) ||
+        equalLower(state.stateCode, stateId),
+    );
+    return foundState || null;
+  }
+
+  findCountyByUrlParams(stateId: string, countyId: string): County | null {
+    const foundState = this.findStateByUrlParams(stateId);
+    if (!foundState) {
+      return null;
+    }
+
+    const foundCounty = this.counties()
+      .filter(county => county.state.fipsCode === foundState.fipsCode)
+      .find(county => equalLower(county.urlSegment, countyId));
+
+    return foundCounty || null;
+  }
+
   all(): Region[] {
     return this.regions;
   }
 
-  states(): Region[] {
-    return this.filterByType(RegionType.STATE);
+  states(): State[] {
+    return this.all().filter(
+      (region): region is State => region.regionType === RegionType.STATE,
+    );
   }
 
-  counties(): Region[] {
-    return this.filterByType(RegionType.COUNTY);
+  counties(): County[] {
+    return this.all().filter(
+      (region): region is County => region.regionType === RegionType.COUNTY,
+    );
   }
 }
 
+function equalLower(a: string, b: string) {
+  return a.toLowerCase() === b.toLowerCase();
+}
+
 const regions = new RegionDB([...states, ...counties]);
 export default regions;
+
+// TODO: Is there a better way to do this? If I pass `null` as default value
+// for the context, we will need to check that the region is not null
+// everywhere (we are checking for that in the LocationRouter)
+const standInRegion: State = new State('name', 'name', '99', 0, '99');
+export const RegionContext = createContext<Region>(standInRegion);
+
+export const useRegion = () => useContext(RegionContext);
```

Comment:
If we think we are going to stick with our "location page" terminology for a while I'd be pretty tempted to name this `useLocationPageRegion()` or similar to make it a bit more explicit what region it is talking about.

Else maybe `useUrlRegion` or something? Idunno.

Reply:
changed to `useLocationPageRegion`